src/chkstat.cpp

```diff
@@ -322,12 +332,20 @@ int Chkstat::processEntries() {
     size_t errors = 0;
 
     if (m_apply_changes && !m_have_proc) {
-        std::cerr << "ERROR: /proc is not available - unable to fix policy violations. Will continue in warn-only mode." << std::endl;
-        errors++;
-        m_apply_changes = false;
+        if (m_allow_no_proc) {
+            std::cerr << "WARNING: /proc is not available, continuing in insecure mode because " << ENVVAR_ALLOW_NO_PROC << " is set." << std::endl;
+        } else {
+            std::cerr << "ERROR: /proc is not available - unable to fix policy violations. Will continue in warn-only mode." << std::endl;
+            errors++;
+            m_apply_changes = false;
+        }
     }
 
     for (const auto& [path, entry]: m_profile_parser.entries()) {
+        if (!m_have_proc && !m_allow_no_proc) {
+            assert(!m_apply_changes);
+        }
```

Comment:
This should go outside of the loop body, these flags won't change any more during the loop iterations.